src/main/java/reactor/core/publisher/FluxPeekFuseable.java

```diff
@@ -258,7 +263,7 @@ public T poll() {
 					throw Exceptions.propagate(Operators.onOperatorError(s, e, v));
 				}
 			}
-			if (v == null && sourceMode == SYNC) {
+			if (v == null && (d || sourceMode == SYNC)) {
```

Comment:
but you read only at that moment, is that due to memory barrier ? In that case what about poll in TakeFuseable  for instance ?